### PR TITLE
rewrite upload images from base64 in database to link on blobStorage

### PR DESCRIPTION
**Pull request by dieie32**

### CODE REVIEW

**Review by kapoliub — DISMISSED** [review]

**Review by jul4ajka — APPROVED** [review]

File: src/components/forms/constructor-form/constructor-form.js [hunk 1]
```diff
@@ -58,23 +66,26 @@ const ConstructorForm = ({ isEdit, editableConstructorElement }) => {
...
 
-  const [materialColors, setMaterialColors] = useState([])
+  const [materialColors, setMaterialColors] = useState([]);
+  const [constructorAvatar, setConstructorAvatar] = useState('');
 
   useEffect(() => {
     if (isEdit) {
-      setConstructorImg(editableConstructorElement.image);
-      setMaterialColors(list.filter(el=>el._id===editableConstructorElement.material._id)[0].colors);
+      setConstructorAvatar(`${IMG_URL}${editableConstructorElement.image}`);
```
Comment (moran711) [inline src/components/forms/constructor-form/constructor-form.js:83]: You don't give possibility to admin to see avatar of photo that he added using input. Are you sure?)

File: src/components/forms/constructor-form/constructor-form.js [hunk 2]
```diff
@@ -58,23 +66,26 @@ const ConstructorForm = ({ isEdit, editableConstructorElement }) => {
...
+      setMaterialColors(
+        list.filter(
```
Comment (moran711) [inline src/components/forms/constructor-form/constructor-form.js:85]: Use lodash function https://lodash.com/docs/4.17.15#filter

File: src/components/forms/constructor-form/constructor-form.js [hunk 3]
```diff
@@ -114,40 +123,46 @@ const ConstructorForm = ({ isEdit, editableConstructorElement }) => {
       default: editableConstructorElement.default || false,
       basePrice: +editableConstructorElement.basePrice[1].value / 100 || 0
     },
-
     onSubmit: () => {
       const constructorElement = createConstructor(values);
```
Comment (moran711) [inline src/components/forms/constructor-form/constructor-form.js:127]: You have `formValues `as a first value that `onSubmit` method accept.

File: src/utils/use-constructor-handlers.js [hunk 4]
```diff
@@ -6,33 +6,30 @@ const { languages } = config;
...
+      {
+        lang: languages[0],
+        value: values.uaName
+      },
+      {
+        lang: languages[1],
+        value: values.enName
+      }
+    ],
+    image: values.image,
```
Comment (moran711) [inline src/utils/use-constructor-handlers.js:26]: You add this image as upload variable in request, and you don't use it. Please remove this

File: src/pages/model/constructor/constructor-page/constructor-page.js [hunk 5]
```diff
@@ -273,7 +273,7 @@ const ConstructorPage = ({ match }) => {
                 </Avatar>
               </ListItemAvatar>
               <ListItemText primary={pattern.name[0].value} />
-              <Avatar src={pattern.constructorImg} variant='rounded' className={styles.avatar}>
+              <Avatar src={IMG_URL + pattern.constructorImg} variant='rounded' className={styles.avatar}>
```
Comment (moran711) [inline src/pages/model/constructor/constructor-page/constructor-page.js:276]: same

File: src/components/forms/pattern-form/pattern-form.js [hunk 6]
```diff
@@ -123,14 +125,25 @@ const PatternForm = ({ pattern, id, isEdit }) => {
       const newPattern = createPattern(values);
 
       if (isEdit && upload instanceof File) {
```
Comment (moran711) [inline src/components/forms/pattern-form/pattern-form.js:127]: You don't check if `uploadConstructorImg` is  instance of `File`

File: src/components/forms/constructor-form/constructor-form.js (hunk 1 above)
Comment (dieie32) [reply src/components/forms/constructor-form/constructor-form.js:83]: image

File: src/pages/model/constructor/constructor-page/constructor-page.js (hunk 5 above)
Comment (dieie32) [reply src/pages/model/constructor/constructor-page/constructor-page.js:276]: image

Comment (dieie32) [reply src/pages/model/constructor/constructor-page/constructor-page.js:276]: image

File: src/components/forms/pattern-form/pattern-form.js (hunk 6 above)
Comment (dieie32) [reply src/components/forms/pattern-form/pattern-form.js:127]: fixed

File: src/components/forms/constructor-form/constructor-form.js (hunk 3 above)
Comment (dieie32) [reply src/components/forms/constructor-form/constructor-form.js:127]: fixed

File: src/utils/use-constructor-handlers.js (hunk 4 above)
Comment (dieie32) [reply src/utils/use-constructor-handlers.js:26]: fixed

File: src/components/forms/constructor-form/constructor-form.js (hunk 2 above)
Comment (dieie32) [reply src/components/forms/constructor-form/constructor-form.js:85]: fixed

**Review by moran711 — APPROVED** [review]

**Review by jul4ajka — APPROVED** [review]

**Review by kapoliub — APPROVED** [review]

**Pull request merged**